packages/framework/src/hooks/useStyles.ts

```diff
@@ -35,40 +34,28 @@ const resolveStyleNames = (
 export const useStyles = <T extends { [key: string]: unknown }>(
   values: T,
 ): CSSProperties | undefined => {
-  const namedStyles = get(values, ["styles", "names"]) as unknown;
-  const classStyles = get(values, ["styles", "className"]) as unknown;
+  const namedStyles = get(values, ["styles", "names"]) as string;
+  const classStyles = get(values, ["styles", "className"]) as string;
 
   const themeContext = useAtomValue(themeAtom);
 
-  const { classStylesArray, nameStylesArray } = useMemo(() => {
-    const stringSplitter = (styles: string): string[] =>
-      // eslint-disable-next-line prefer-named-capture-group
-      styles.split(/(\s+)*(\${.+?})(\s+)*/g).filter(Boolean);
-    return {
-      classStylesArray: isString(classStyles)
-        ? stringSplitter(classStyles)
-        : [],
-      nameStylesArray: isString(namedStyles) ? stringSplitter(namedStyles) : [],
-    };
-  }, [classStyles, namedStyles]);
-
-  const {
-    classStylesArray: classStylesArrayEval,
-    nameStylesArray: namedStylesArrayEval,
-  } = useEvaluate({
-    classStylesArray,
-    nameStylesArray,
-  });
+  const { classStyles: classStylesEval, namedStyles: namedStylesEval } =
+    useEvaluate({
+      classStyles: `\${\`${classStyles || ""}\`}`,
+      namedStyles: `\${\`${namedStyles || ""}\`}`,
+    });
 
   const styleNames = compact([
-    ...namedStylesArrayEval,
-    ...classStylesArrayEval.map((className) => `.${className}`),
+    ...(classStylesEval.trim()
+      ? classStylesEval.split(" ").map((className) => `.${className}`)
+      : []),
+    ...[namedStylesEval.trim()],
```

Comment:
Don't need to split here?

Reply:
We need to assign `.` prefix  to each classname, and this is the only reason to split here